source/common/http/http1/conn_pool.cc

```diff
@@ -220,6 +229,16 @@ void ConnPoolImpl::onResponseComplete(ActiveClient& client) {
 
 void ConnPoolImpl::onUpstreamReady() {
   upstream_ready_enabled_ = false;
+  auto it = delayed_clients_.begin();
+  while (it != delayed_clients_.end()) {
+    ActiveClient& client = **it;
+    it++;
```

Comment:
This pattern is rear. Better to add comment that `moveBetweenLists` will invalidate the it if don't move forward

Reply:
Done